TempleEight/docs/guide/getting-started.md

```diff
@@ -0,0 +1,281 @@
+
+### Service Blocks
+
+```
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+The next thing to add is a Service Block.
+Each service block in a Templefile defines one generated microservice. 
+
+The opening statement `TestService: service` defines a new service, named `ExampleService`, which is the name Temple will use to refer to this service.
+
+Service block can contain two kinds of declarations: *Property definitions* and *metadata*. 
+
+Property declarations tell us about the kind of data belonging to the entity this service describes. 
+For example, here we say that `ExampleService` has a `foo`  which is a `string` property, and a `bar`, which is an `int` property. 
+Property declarations can either be a [*primitive* datatype](primitives), or be a [*foreign key*](cross-service-coms) to another service.
+
+Every service in Temple also has an implicit `id` property, which assigns a `UUID` to each entity stored in a service, used for foreign keys and authentication.
+
+> Note: Variable names (project names, service names and property names) can clash with reserved keywords in the languages Temple generates into.
+> To mitigate this, the Temple CLI will automatically rename any conflicting variables to `$projectName-$variableName`.
+
+
+All together, our example Templefile is:
+
+```
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+## Running The Example
+
+Now that we have created our example Templefile, save it in a file named `example.temple`, in our `temple-tutorial` directory created earlier.
+
+### Validating the Templefile
+
+The Temple CLI contains a tool for validating your Templefile before generating any code. We can invoke this tool by running:
+
+```
+~/Documents/temple-tutorial ❯❯❯ temple validate example.temple
+Templefile validated correctly
+```
+
+This means our Templefile is syntactically correct, and we can begin to generate all the code we need.
+
+### Generating Code
+
+At this point, we can begin to generate some code. Running the following will begin generating code: 
+
+```
+~/Documents/temple-tutorial ❯❯❯ temple generate example.temple
+```
+
+The first thing we'll see is the CLI prompting us to give some further generation details that are specific to the metadata selections made in our Templefile. 
+
+```
+~/Documents/temple-tutorial ❯❯❯ temple generate example.temple
+What should the Go module name be? (expected format "github.com/username/repo")
+```
+
+We need to enter the name of the Golang module we'll be generating into for our project, this can be anything, but standards dictate it should be the GitHub URL of where this project will be hosted. 
+
+For this example enter `github.com/temple/tutorial`. 
+
+```
+~/Documents/temple-tutorial ❯❯❯ temple generate example.temple
+What should the Go module name be? (expected format "github.com/username/repo")
```

Comment:
I wonder if this could also be specified within the Templefile?

Reply:
I think it would be have to be an argument to the #language block or something, since we only need it specifically when go is the chosen lang